crates/egui/src/memory.rs

```diff
@@ -151,6 +155,32 @@ impl FocusDirection {
     }
 }
 
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "persistence", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "persistence", serde(default))]
+pub struct LayerTransform {
+    pub translation: Vec2,
+    pub scale: f32,
+}
+
+impl Default for LayerTransform {
+    fn default() -> Self {
+        Self {
+            translation: Vec2::default(),
+            scale: 1.0,
+        }
+    }
+}
+
+impl LayerTransform {
+    pub(crate) fn apply(&self, rect: Rect) -> Rect {
```

Comment:
we should name it for what it transforms, so we can also transform positions

```suggestion
    #[inline]
    pub fn mul_rect(&self, rect: Rect) -> Rect {
```